pkgs/top-level/octave-packages.nix

```diff
@@ -0,0 +1,92 @@
+# This file contains the GNU Octave add-on packages set.
+# Each attribute is an Octave library.
+# Expressions for the Octave libraries are supposed to be in `pkgs/development/octave-modules/<name>/default.nix`.
+
+{ pkgs
+, lib
+, stdenv
+, fetchurl
+, newScope
+, octave
+, lapack, blas
+, gfortran
+, autoreconfHook
+, python3
+, python3Packages
+, jdk
+, gnuplot
+, texinfo
+, nettle
+}:
+
+let
+
+  isOctaveFull = octave.enableQt;
+  isOctaveJIT = octave.enableJIT;
+
+  buildOctaveLibrary = callPackage ../development/interpreters/octave/mk-octave-derivation.nix {
+    inherit pkgs lib stdenv;
+    namePrefix = "${octave.pname}-${octave.version}";
+    inherit octave;
+  };
+
+  callPackage = pkgs.newScope {
+    inherit (pkgs) lib stdenv;
+    inherit buildOctaveLibrary;
+    inherit fetchurl;
+    inherit gnuplot texinfo;
+  };
+
+in rec {
+
+  control = callPackage ../development/octave-modules/control {
+    gfortran = gfortran;
+    autoreconfHook = autoreconfHook;
+    lapack = lapack;
+    blas = blas;
+  };
+
+  io = callPackage ../development/octave-modules/io {
+    enableJava = true;
```

Comment:
This should be conditioned based on whether octave is built with java.

Reply:
This is a bit of a moot point, since [io-2.4.13   Release Date: 2019-10-15](https://sourceforge.net/p/octave/io/ci/default/tree/NEWS), when the explicit dependency on java was removed, as java support should be built into octave.